File: client/src/tis-todo/src/app/models/todo-item.ts
import { TodoCategory } from "./todo-category";

export interface TodoItem {
    id: string;
    title: string;
    description: string;
    createdAt: string;
    lastUpdatedAt: string;
    dueAt: string;
    isComplete: boolean;
    creatorId: string;
    category: string;
}

File: server/src/core/TIS.Todo.Domain/Models/TodoItem.cs
using System;

namespace TIS.Todo.Domain.Models
{
    public class TodoItem : BaseEntity
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime LastUpdatedAt { get; set; }
        public DateTime? DueAt { get; set; }
        public bool IsComplete { get; set; }
        public string UserId { get; set; }
        public AppUser User { get; set; }
        public string Category { get; set; }
    }
}